test/AST-Quake/single_qubit_ctor.cpp

```diff
@@ -6,7 +6,7 @@
  * the terms of the Apache License 2.0 which accompanies this distribution.    *
  ******************************************************************************/
 
-// RUN: cudaq-quake %cpp_std %s | FileCheck %s
+// RUN: cudaq-quake %cpp_std %s | cudaq-opt | FileCheck %s
```

Comment:
why this change?

Reply:
The latest update to `qubit_qis.h` takes all quantum operation parameters by reference now. Previously, any scalar rotation parameters were passed by value. This change caused the bridge to produce `cc.ptr<f64>` for the parameter instead of just `f64`. This also lead to some cases of `alloca : ptr<f64>` and so I made a canonicalization pass to remove those patterns in an effort to minimize changes to the FileCheck tests. Interested in other opinions here though.

Reply:
Curious, but why change to use references? That seems like it will be inefficient for C++ and generate stack clutter (which we will have to manage better), but I'm assuming we're doing this because of Python.

It should be possible to introduce an intrinsic and/or runtime C++ entry point that will convert values to stack slots, bounds the lifetime of these ephemeral references, etc.